powerbi-docs/developer/visuals/license-api.md

```diff
@@ -0,0 +1,178 @@
+                    if (value) {
+                        this.isIconDisplayed = true;
+                    }
+                }).catch((err) => {
+                    console.log('ERROR', err);
+                })
+            }
+        }).catch((err) => {
+            this.hasServicePlans = undefined;
+            console.log(err);
+        });
+```
+
+## Notify the user that the required licenses are missing
+
+Power BI platform provides several out of the box experiences that can be used to notify:
+* Licenses should be purchased in order to enjoy full visual's capabilities
+* Particular visual's feature is blocked due to missing licenses
+* Entire visual is blocked due to missing licenses
+* Entire visual is blocked because the Power BI environment in use doesn't support license management\enforcement
+
+```typescript
+export interface IVisualLicenseManager {
+        notifyLicenseRequired(notificationType: LicenseNotificationType): IPromise<boolean>;
+        notifyFeatureBlocked(tooltip: string): IPromise<boolean>;
+        clearLicenseNotification(): IPromise<boolean>;
+    }
+```
+
+
+### General icon to indicate that the required licenses are missing
+
+Use `notifyLicenseRequired` call with `LicenseNotificationType.General` to display an icon as part of the visual's container.  
+Once triggered, the icon will be preserved throughout the visual's lifetime until `clearLicenseNotification` is called.
+
+> [!NOTE]
+> The `LicenseNotificationType.General` is only supported in Power BI Edit scenarios. Calling this when the report is in Read mode or dashboard scenario will not apply the icon and will return `false` in the call's response.   
+
+Example of the visual display containing the "licenses are required" general icon:
+
+>[!div class="mx-imgBorder"]
+>![visual display containing the "licenses are required" general icon.](media/licensing/generalIcon.png)
+
+
+>[!div class="mx-imgBorder"]
+>![visual display containing the "licenses are required" general icon.](media/licensing/generalIconExpand.png)
+
+
+### Overlay the visual's display with a notification on missing licenses
+
+Use `notifyLicenseRequired` call with `LicenseNotificationType.VisualIsBlocked` to overlay the visual's display with a notification that visual is blocked since required licenses were found missing.  
+Once triggered, the icon will be preserved throughout the visual's lifetime until `clearLicenseNotification` is called.
+
+Example of the visual display containing the "visual blocked" notification:
+
+>[!div class="mx-imgBorder"]
+>![visual display containing the "visual blocked" notification.](media/licensing/blockedVis.png)
+
+
+### Overlay the visual's display with a notification on licenses unsupported environment
+
+Use `notifyLicenseRequired` call with `LicenseNotificationType.UnsupportedEnv` to overlay the visual's display with a notification that visual is blocked since the Power BI in use doesn't support licenses management\enforcement.  
```

Comment:
Please mention that the unsupported env blocker can be displayed only if it's an unsupported env.